File: Daily/February/LeetCode1189.cc
#include<vector>
#include<unordered_map>
#include<algorithm>
using namespace std;

class Solution {
public:
    int maxNumberOfBalloons(string text) {
       
       /**
        * 1.统计每个字母出现的次数
        * 2. l和o需要2个,所以要除以2.
        * 3.求出最小值即可
        */
       vector<int> cnt(5);

       for(auto &context:text){
           if(context == 'b'){
               cnt[0]++;
           }else if(context == 'a'){
               cnt[1]++;
           }else if(context == 'l'){
               cnt[2]++;
           }else if(context == 'o'){
               cnt[3]++;
           }else if(context == 'n'){
               cnt[4]++;
           }
       }
       cnt[2] /= 2;
       cnt[3] /= 2;

       return *min_element(cnt.begin(),cnt.end());
    }
};

File: Daily/February/LeetCode1189.go
package february

func maxNumberOfBalloons(text string) int {

	cnt := make([]int, 5)
	for i := 0; i < len(text); i++ {
		if text[i] == 'b' {
			cnt[0]++
		} else if text[i] == 'a' {
			cnt[1]++
		} else if text[i] == 'l' {
			cnt[2]++
		} else if text[i] == 'o' {
			cnt[3]++
		} else if text[i] == 'n' {
			cnt[4]++
		}
	}
	cnt[2] /= 2
	cnt[3] /= 3
}

func min(begin, end int) int {
	minNum :=
	for begin := 0; begin < end; begin++ {

	}
}
